main/src/ca/uwaterloo/flix/language/phase/Resolver.scala

```diff
@@ -3165,13 +3167,31 @@ object Resolver {
   /**
     * Returns the class reflection object for the given `className`.
     */
-  private def lookupJvmClass(className: String, loc: SourceLocation)(implicit flix: Flix): Result[Class[_], ResolutionError with Recoverable] = try {
+  private def lookupJvmClass(className: String, loc: SourceLocation, ns0: Name.NName, root: NamedAst.Root)(implicit flix: Flix): Result[Class[_], ResolutionError with Recoverable] = try {
     // Don't initialize the class; we don't want to execute static initializers.
     val initialize = false
     Result.Ok(Class.forName(className, initialize, flix.jarLoader))
   } catch {
-    case ex: ClassNotFoundException => Result.Err(ResolutionError.UndefinedJvmClass(className, ex.getMessage, loc))
-    case ex: NoClassDefFoundError => Result.Err(ResolutionError.UndefinedJvmClass(className, ex.getMessage, loc))
+    case ex: ClassNotFoundException => lookUpJvmClassInImport(className, loc, ns0, root)
+    case ex: NoClassDefFoundError => lookUpJvmClassInImport(className, loc, ns0, root)
+  }
+
+  private def lookUpJvmClassInImport(className: String, loc: SourceLocation, ns0: Name.NName, root: NamedAst.Root)(implicit flix: Flix): Result[Class[_], ResolutionError with Recoverable] = {
```

Comment:
I am surprised this method is necessary. I would expect something like it to already exist?

Reply:
Actually, instead of doing this, can you just do what InvokeStaticMethod2 does? Thats going to be the future anyway.

Reply:
true maybe I just haven't looked around enough :D